### PR TITLE
chore: add release docs and handle newer Terraform version

### PR DESCRIPTION
**Pull request by Shabirmean**
#### Description
- As raised in https://github.com/GoogleCloudPlatform/anthos-samples/pull/318 we don't have proper documentation on what to look into when releasing the repo. 
- Also I have updated the docker image version used by the [cft framework](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/master/infra/build/Makefile#L35) we use for integration tests. This latest version of theirs uses [terraform 1.1.9](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/master/infra/build/Makefile#L16)

#### Change summary
- Add a releasing guidelines doc
- Mention the things discussed [in this PR](https://github.com/GoogleCloudPlatform/anthos-samples/pull/318#discussion_r882088614)
- Update all places where Terraform version is relavant

#### Related PRs/Issues
- https://github.com/GoogleCloudPlatform/anthos-samples/pull/318

### CODE REVIEW

File: .github/gh_runner_dependencies.sh [hunk 1]
```diff
@@ -46,7 +46,8 @@ tfenv install 1.0.0
 tfenv install 1.0.1
 tfenv install 1.0.6
 tfenv install 1.1.3
-tfenv use 1.1.3
+tfenv install 1.2.0
+tfenv use 1.2.0
```
Comment (ckim328) [inline .github/gh_runner_dependencies.sh:50]: do we also manually add these?

File: .github/release.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Releasing Guidelines
```
Comment (ckim328) [inline .github/release.md:1]: this is 🔥 looks great

File: .github/gh_runner_dependencies.sh (hunk 1 above)
Comment (Shabirmean) [reply .github/gh_runner_dependencies.sh]: So I was lazy to SSH into the VMs and set it up. So I added the install commands as part of the CI yamls and removed them after letting it run once: https://github.com/GoogleCloudPlatform/anthos-samples/pull/319/commits/c991267dfe7be7c3ed15db597038360f7e779af2

I am not sure which yaml files you are referring to here!!!

Comment (ckim328) [reply .github/gh_runner_dependencies.sh]: (sorry i should've specified!)
these yaml files: 
- .github/workflows/ci_main_branch.yaml
- .github/workflows/ci_main_branch.yaml

i see that the tfenv uses 1.1.3 in the above files in the PR

**Review by ckim328 — APPROVED** [review]
LGTM 👍

**Pull request merged**